### ISSUE TITLE
Not using a library or ES6 `fetch` to perform network requests in JS

### ISSUE THREAD

**Issue by olivierobert** [nil1511/google-keywords#7]
![image](https://user-images.githubusercontent.com/696529/83398757-2dab3c80-a42a-11ea-8d31-5909430aab7e.png)

While the current implementation works, it's unlikely it would be used in a larger application due to the required boilerplate code. In addition, placing the code directly in the pack file make this code evaluated on every page while it should be scoped to the dashboard page only.

In addition, `getElementsByClassName` should preferably be replaced by the more modern and support `querySelector`.

**Comment by nil1511** [comment]
> While the current implementation works, it's unlikely it would be used in a larger application due to the required boilerplate code. In addition, placing the code directly in the pack file make this code evaluated on every page while it should be scoped to the dashboard page only.

This code will be part of `dashboard.js` and is loaded only for dashboard view.



> In addition, getElementsByClassName should preferably be replaced by the more modern and support querySelector.

Yes. I will update this.

**Comment by olivierobert** [comment]
> This code will be part of dashboard.js and is loaded only for dashboard view.

Correct. I missed that part. But does this mean that a new manifest file would be needed for each screen? Which other way would you implement to load specific functionality for each screen?

**Comment by nil1511** [comment]
We should have a `[screen].js` for each screen and a (vendor stuff,global) js like `application.js`. In `[screen].js` we can use dynamic import to load js semi-shared code(shared.js) and specific code(specific.js) and reap the benefit of code-splitting and decreasing the bytes sent to the user.

`import('/shared.js')... import('/specific.js')...` in `[screen1].js`
`import('/shared.js')...` in `[screen2].js`

**Comment by olivierobert** [comment]
In the current state of the application, I would expect to be more ready for when the project has more than 2 screens. If there was more than one JS functionality (apart from the form upload), I would expect the code to:
- Be moved to specific files e.g. `form_upload.js` (which is then imported into the manifest file)
- Not to have everything in the global scope to avoid naming clashes 😭

```
const FormUpload = () {
  const handleFileChange = () => {};
  #....
}

FormUpload()
```
